src/devices/IBT_2X_json.h

```diff
@@ -7,22 +7,22 @@ const char* _IBT_2X_json = "{\"brand\":\"Inkbird\",\"model\":\"iBBQ\",\"model_id
    "condition":["name", "index", 0, "iBBQ","&","manufacturerdata", ">=", 24, "index", 0, "0000","&","manufacturerdata", "<=", 28, "index", 0, "0000"],
    "properties":{
       "tempc":{
-         "condition":["manufacturerdata", 0, "00000000", "&", "manufacturerdata", 16, "!", "f6ff"],
```

Comment:
is it voluntary to impact IBT_2X conditions? I'm not understanding why

Reply:
Actually there still is an issue with the IBT_2X, and also BT_4XS decoder, with them needing to have their compressed const char* properly done. It was thanks to the IBT_2X that the multi byte-index issue became apparent. With fixing those two decoders I will reintroduce the NOT for their high/probe not connected temperatures.

Reply:
I can remove these changes, I used them from @DigiH's fork to test what he was testing. I forgot about them when I made the PR and should be separated to a new PR.

Reply:
okay got it

Reply:
The fixes which I preprepared for the IBT_2X are at

https://github.com/DigiH/decoder/blob/not-fixes/src/devices/IBT_2X_json.h

which now should finally have 

"condition":["manufacturerdata", 0, "!", "00000000", "&", "manufacturerdata", 16, "!", "f6ff"],

for tempc and tempc2. Which I think is the first double NOT AND condition

Reply:
Updated with @DigiH's fixes as a separate commit, squashed all the others.

Reply:
>`  "&", "manufacturerdata", 16, "!", "f6ff"],`

I think this check and the others like this one are missing on the branch isn'it?

Reply:
These were the changes I wanted to submit once this was merged, with additional tests for the previous problematic IBT-2X. Maybe @h2zero can grab this an included it here with this PR.

https://github.com/DigiH/decoder/commit/6c4556091d5f15e1eb9ee24718835c60c492f66e

Reply:
@h2zero could you either remove the IBT_2X changes or pick the changes from here please https://github.com/DigiH/decoder/commit/0fb6a9392e8164ff42c513fcc7ed2ffd6e1e0bbe

Reply:
Probably easiest if I cherry pick this over, if that's ok.

Reply:
Yes I think so

Reply:
Done

Reply:
Cool, thanks! I wasn't sure what specific changes you were referring to, I only used the ones in the link provided.